src/pools/selection-strategies.ts

```diff
@@ -106,6 +110,89 @@ class LessRecentlyUsedWorkerChoiceStrategy<
   }
 }
 
+/**
+ * Selects the next worker with a fair share tasks scheduling algorithm.
+ *
+ * @template Worker Type of worker which manages the strategy.
+ * @template Data Type of data sent to the worker. This can only be serializable data.
+ * @template Response Type of response of execution. This can only be serializable data.
+ */
+class FairShareChoiceStrategy<Worker extends IWorker, Data, Response>
+  implements IWorkerChoiceStrategy<Worker> {
+  /**
+   *  Worker last virtual task execution end timestamp.
+   */
+  private workerLastVirtualTaskFinishTimestamp: Map<Worker, number>
```

Comment:
I think you can simplify this to directly initialize it

```suggestion
  private workerLastVirtualTaskFinishTimestamp: Map<Worker, number> = new Map<Worker, number>()
```

Reply:
Hum, I think the map need to be reset at object creation to ensure a fresh start when switching strategy at runtime but since strategy are not singleton, that's fine to do put it there.